### PR TITLE
Time machine feature

### PR DESCRIPTION
**Pull request by VB-at-Bis**
- added mock.now() and related function/mutations;
- wip front-end changes;

### CODE REVIEW

**Comment by matthieu-foucault** [comment]
@VB-at-Bis I see 120+ changed files. Sounds like git issues?

File: app/components/CalendarIcon.tsx [hunk 1]
```diff
@@ -0,0 +1,66 @@
+import styled from 'styled-components';
```
Comment (marcellmueller) [inline app/components/CalendarIcon.tsx:1]: I was going to refactor the `datePickerWidget` but then decided against it, though I kept this pulled out if anyone is wondering what this is about.

Comment (marcellmueller) [inline app/components/CalendarIcon.tsx:1]: I was going to refactor the `datePickerWidget` but then decided against it, though I kept this pulled out if anyone is wondering what this is about.

File: app/formSchema/pages/timeMachine.ts [hunk 2]
```diff
@@ -0,0 +1,19 @@
+const timeMachine = {
+  timeMachine: {
+    title: 'Time Mashine',
```
Comment (marcellmueller) [inline app/formSchema/pages/timeMachine.ts:3]:
```suggestion
    title: 'Time Machine',
```

File: app/config.js [hunk 3]
```diff
@@ -144,6 +144,18 @@ const config = convict({
...
+    format: Boolean,
+    default: false,
+    env: 'ENABLE_MOCK_TIME',
+  },
+  ENABLE_MOCK_COOKIES: {
+    doc: 'Enable DB Mocks Cookies',
+    format: Boolean,
+    default: false,
+    env: 'ENABLE_MOCK_COOKIES',
+  },
```
Comment (matthieu-foucault) [inline app/config.js:158]: We should ensure that these options are never set in the prod environment. See 
https://github.com/bcgov/cas-cif/blob/359d5dcacfd88874aa70aedf5550f32610c6750c/app/config.js#L220 for an example

File: db/sqitch.plan [hunk 4]
```diff
@@ -23,3 +25,4 @@ tables/application_status 2022-07-08T13:27:08Z Marcel Mueller <marcel@button.is>
 tables/attachment [tables/application] 2022-07-07T23:19:04Z Marcel Mueller <marcel@button.is> # add table attachment
 functions/open_intake 2022-08-16T13:48:54Z Anthony Bushara <anthony@button.is> # Function to display the current open intake
 mutations/submit_application 2022-08-23T17:18:10Z Matthieu Foucault <matthieu@button.is> # create a mutation to submit an application
+
```
Comment (matthieu-foucault) [inline db/sqitch.plan:28]: The changes to this file should be undone as we have a separate sqitch plan for it

File: app/components/Layout.tsx [hunk 5]
```diff
@@ -66,6 +65,8 @@ const Layout: React.FC<Props> = ({ children, session, title }) => {
       </Head>
       <Navigation isLoggedIn={isLoggedIn} />
       <StyledMain>{children}</StyledMain>
+      <TimeTravel />
+
```
Comment (matthieu-foucault) [inline app/components/Layout.tsx:69]: This should be conditionally enabled, based on the `ENABLE_MOCK_TIME` setting

**Review by matthieu-foucault — DISMISSED** [review]

File: app/formSchema/pages/timeMachine.ts (hunk 2 above)
Comment (marcellmueller) [reply app/formSchema/pages/timeMachine.ts:3]: Actually I think this page just needs to be removed as I removed the rest from the schema sorry.

File: app/components/Layout.tsx (hunk 5 above)
Comment (VB-at-Bis) [reply app/components/Layout.tsx]: Implemented

File: app/config.js (hunk 3 above)
Comment (VB-at-Bis) [reply app/config.js:158]: Implemented

File: app/formSchema/pages/timeMachine.ts (hunk 2 above)
Comment (VB-at-Bis) [reply app/formSchema/pages/timeMachine.ts]: Page removed

File: db/sqitch.plan (hunk 4 above)
Comment (VB-at-Bis) [reply db/sqitch.plan]: Moved to separate sqitch plan

File: app/components/Dashboard/Table.tsx [hunk 6]
```diff
@@ -105,19 +107,17 @@ const Table = ({ applications }: Props) => {
 
             const lastEditedIndex = formPages.indexOf(lastEditedPage) + 1;
 
-            const intakeClosingDate = intakeByIntakeId?.closeTimestamp;
-            const isIntakeClosed = intakeClosingDate
-              ? Date.parse(intakeClosingDate) < Date.now()
-              : false;
+            const mockDate = cookie.get('mocks.mocked_timestamp');    
+            const isIntakeOpen = checkIntakeValidity(intakeByIntakeId?.openTimestamp, intakeByIntakeId?.closeTimestamp, mockDate);
```
[the remark this answers is not in the document]
Comment (VB-at-Bis) [reply app/components/Dashboard/Table.tsx:111]: Graphql response has only intake open date and intake closing date, but the status of the application depends on the status of the intake which is calculated here based on the cookie. Status is need to enable/disable Edit button for the draft applications

**Comment by VB-at-Bis** [comment]
Closing this PR - please see https://github.com/bcgov/CONN-CCBC-portal/pull/633 instead.

**Pull request closed without merging**